src/http/server_context.rs

```diff
@@ -121,4 +82,84 @@ impl ServerContext {
       env_vars: Arc::new(env),
     }
   }
+
+  #[allow(clippy::too_many_arguments)]
+  fn check_resolver(
+    field: &mut blueprint::FieldDefinition,
+    expr: &Expression,
+    bt: Batch,
+    universal_http_client: Arc<dyn HttpClient>,
+    http2_only_client: Arc<dyn HttpClient>,
+    http_data_loaders: &mut Vec<DataLoader<DataLoaderRequest, HttpDataLoader>>,
+    gql_data_loaders: &mut Vec<DataLoader<DataLoaderRequest, GraphqlDataLoader>>,
+    grpc_data_loaders: &mut Vec<DataLoader<grpc::DataLoaderRequest, GrpcDataLoader>>,
```

Comment:
instead of passing all that options to the function consider this:
- make this function associated fn that accepts self
- self will have most of that options already
- in new functions first create the ServerContext then call this function to validate

Reply:
yes, but we will keep this code private inside constructor function to prevent misusage